src/Juvix/Backends/Michelson/Parameterisation.hs

```diff
@@ -0,0 +1,65 @@
+module Juvix.Backends.Michelson.Parameterisation where
+
+import qualified Juvix.Core.Erased.Types as C
+import qualified Juvix.Core.Types as C
+import Juvix.Library
+import qualified Michelson.Untyped as M
+import Text.ParserCombinators.Parsec
+import qualified Text.ParserCombinators.Parsec.Token as Token
+import Prelude (String)
+
+type Term = C.Term PrimVal
+
+type Type = C.Type PrimTy
+
+type Value = M.Value' M.ExpandedOp
+
+type Op = M.ExpandedOp
+
+data PrimTy
+  = PrimTy M.Type
+  deriving (Show, Eq, Generic)
+
+data PrimVal
+  = PrimConst (M.Value' Op)
+  | PrimPair
+  | PrimFst
+  | PrimSnd
+  -- TODO: Add all Michelson instructions which are functions.
+
+  deriving (Show, Eq, Generic)
+
+-- TODO: Add rest of primitive values.
+typeOf ∷ PrimVal → [PrimTy]
+typeOf (PrimConst v) = [PrimTy (M.Type (constType v) "")]
+
+constType ∷ M.Value' Op → M.T
+constType v =
```

Comment:
Add an error for the missing cases?

Reply:
They will all be filled out later, this function can't fail, I just haven't done it yet.